cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/GlobalParam.scala

```diff
@@ -0,0 +1,304 @@
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+trait GlobalKey[T] {
+  val name: String
+  val isServiceParam: Boolean
+}
+
+case object OpenAIDeploymentNameKey extends GlobalKey[String]
+  {val name: String = "OpenAIDeploymentName"; val isServiceParam = true}
+case object TestParamKey extends GlobalKey[Double] {val name: String = "TestParam"; val isServiceParam = false}
+case object TestServiceParamKey extends GlobalKey[Int]
+  {val name: String = "TestServiceParam"; val isServiceParam = true}
+
+object GlobalParams {
+  private val ParamToKeyMap: mutable.Map[Any, GlobalKey[_]] = mutable.Map.empty
+  private val GlobalParams: mutable.Map[GlobalKey[_], Any] = mutable.Map.empty
+
+  private def boxedClass(c: Class[_]): Class[_] = {
```

Comment:
What happens in case where type isn’t matched?

Reply:
Anything that doesn't match one of Integer, Long, Double, Float, or Boolean is just returned as is

Reply:
Removed the boxedClass riff-raff